textselect.cpp

```diff
@@ -150,11 +183,53 @@ void TextSelect::handleMouseDown(const ImVec2& cursorPosStart) {
     const float textHeight = ImGui::GetTextLineHeightWithSpacing();
     ImVec2 mousePos = ImGui::GetMousePos() - cursorPosStart;
 
-    // Get Y position of mouse cursor, in terms of line number (clamped to the valid range)
-    std::size_t y = static_cast<std::size_t>(std::min(std::max(std::floor(mousePos.y / textHeight), 0.0f), static_cast<float>(numLines - 1)));
+    std::string_view currentLine;
+    std::size_t x, y;
+    if (enableWordWrap) {
+        ImGuiWindow *window = ImGui::GetCurrentWindow();
+
+        const float wrapWidth = ImGui::CalcWrapWidthForPos(window->DC.CursorPos, 0);
+
+        // Find the index of the line under the cursor and its size.
+        float accumulatedHeight = 0;
+        ImVec2 currentLineSize = {};
+        y = numLines - 1;
+        for (std::size_t i = 0; i < numLines - 1; ++i) {
+            std::string_view line = getLineAtIdx(i);
+            currentLineSize = ImGui::CalcTextSize(line.data(), line.data() + line.size(), false, wrapWidth);
+
+            if (mousePos.y < accumulatedHeight + currentLineSize.y) {
+                y = i;
+                break;
+            }
+
+            accumulatedHeight += currentLineSize.y + ImGui::GetCurrentContext()->Style.ItemSpacing.y;
+        }
+        if (y >= numLines)
+            return;
+
+        currentLine = getLineAtIdx(y);
+
+        // Calculate index of the sub-line in the current line.
+        std::size_t localWrappedY = static_cast<std::size_t>(std::min(std::max(std::floor((mousePos.y - accumulatedHeight) / textHeight), 0.0f), std::round(currentLineSize.y / textHeight) - 1));
+
+        ImFont *font = ImGui::GetCurrentContext()->Font;
+
+        auto subLines = wrapText(currentLine, wrapWidth, font);
+        auto currentSubLine = subLines[localWrappedY];
+
+        x = currentSubLine.data() - currentLine.data() + getCharIndex(currentSubLine, mousePos.x);
+    } else {
+        // Get Y position of mouse cursor, in terms of line number (clamped to the valid range)
+        y = static_cast<std::size_t>(std::min(std::max(std::floor(mousePos.y / textHeight), 0.0f), static_cast<float>(numLines - 1)));
+        if (y >= numLines)
+            return;
```

Comment:
This is another place where we can add braces.

```suggestion
        if (y >= numLines) {
            return;
        }
```